vispy/scene/widgets/widget.py

```diff
@@ -115,6 +118,79 @@ def size(self, s):
         self.events.resize()
 
     @property
+    def width(self):
+        """The width of this widget"""
+        return self._size[0]
+
+    @width.setter
+    def width(self, width):
+        self.size = (width, self._size[1])
+
+    @property
+    def min_width(self):
+        return self._width_limits[0]
+
+    @min_width.setter
+    def min_width(self, min_width):
+        if min_width is None:
+            self._width_limits[0] = None
+            return
+
+        self._width_limits[0] = float(min_width)
+
+    @property
+    def max_width(self):
+        return self._width_limits[1]
+
+    @max_width.setter
+    def max_width(self, max_width):
+        if max_width is None:
+            self._width_limits[1] = None
+            return
+
+        self._width_limits[1] = float(max_width)
+
+    def fixed_width(self, fixed_width):
+        self.min_width = self.max_width = fixed_width
+
+    @property
+    def height(self):
+        """The height of this widget"""
+        return self._size[1]
+
+    @height.setter
+    def height(self, height):
+        self.size = (self._size[0], height)
+
+    @property
+    def min_height(self):
+        return self._height_limits[0]
+
+    @min_height.setter
+    def min_height(self, min_height):
+        if min_height is None:
+            self._height_limits[0] = None
+            return
+
+        self._height_limits[0] = float(min_height)
+
+    @property
+    def max_height(self):
+        return self._height_limits[1]
+
+    @max_height.setter
+    def max_height(self, max_height):
+        if max_height is None:
+            self._height_limits[1] = None
+            return
+
+        self._height_limits[1] = float(max_height)
+
+    def fixed_height(self, fixed_height):
+        self.min_height = fixed_height
+        self.max_height = fixed_height
```

Comment:
I think the names should probably be `height_min`, `height_max`, `height_fixed`, `height_stretch`, and `width_min`, `width_max`, `width_fixed`, and `width_stretch`. Right now we have some of them split by `width` / `height` and others grouped (like `stretch`). We should be consistent.